pytensor/link/mlx/dispatch/math.py

```diff
@@ -0,0 +1,250 @@
+import mlx.core as mx
+
+from pytensor.link.mlx.dispatch import mlx_funcify, mlx_typify
+from pytensor.link.mlx.dispatch.core import convert_dtype_to_mlx
+from pytensor.scalar import Softplus
+from pytensor.scalar.basic import (
+    AND,
+    EQ,
+    GE,
+    GT,
+    LE,
+    LT,
+    NEQ,
+    OR,
+    Abs,
+    Add,
+    Cast,
+    Cos,
+    Exp,
+    Log,
+    Log1p,
+    Mul,
+    Neg,
+    Pow,
+    ScalarMaximum,
+    ScalarMinimum,
+    Sign,
+    Sin,
+    Sqr,
+    Sqrt,
+    Sub,
+    Switch,
+    TrueDiv,
+)
+from pytensor.scalar.math import Sigmoid
+from pytensor.tensor.elemwise import Elemwise
+from pytensor.tensor.math import Dot
+
+
+@mlx_typify.register(int)
+@mlx_typify.register(float)
+def mlx_typify_python_scalar(data, **kwargs):
+    return mx.array(data)
+
+
+@mlx_funcify.register(Dot)
+def mlx_funcify_Dot(op, **kwargs):
+    def dot(x, y):
+        return mx.matmul(x, y)
+
+    return dot
+
+
+@mlx_funcify.register(Elemwise)
+def mlx_funcify_Elemwise(op, **kwargs):
```

Comment:
Like CAReduce it should have a second level dispatch. Also we need to enforce the `runtime_broadcastable` checks (same in Alloc). And we shoud have a default implementation for that second level dispatch that tries to use `getattr(MLX, "func_name")` similar to how JAX does it already.